### PR TITLE
Added expired confirmation_token logic and corresponding error as res…

### PR DESCRIPTION
**Pull request by icecream78**
Add logic for confirmation token expirity according to issue #25  .
To verify function was added hardcoded check for confirmation token/recovery token send time checking (24 hours) + add few test to it

### CODE REVIEW

**Comment by awalias** [comment]
love it @icecream78 ! 

One question: If confirmation_token is expired after 24 hours, is it possible to generate another via sign up route? 🤔 

just need to verify that people won't be locked out permanently if they miss the 24 hour confirmation time

**Comment by icecream78** [comment]
@awalias sure, you can regenerate token again via separate call of sign up route

File: api/mail.go [hunk 1]
```diff
@@ -36,7 +36,8 @@ func sendInvite(tx *storage.Connection, u *models.User, mailer mailer.Mailer, re
 		return errors.Wrap(err, "Error sending invite email")
 	}
 	u.InvitedAt = &now
-	return errors.Wrap(tx.UpdateOnly(u, "confirmation_token", "invited_at"), "Database error updating user for invite")
+	u.ConfirmedAt = &now
```
Comment (awalias) [inline api/mail.go:39]: I don't think we should add confirmed at until the user has clicked the confirmation link, or if auto-confirm env var is set to true

Comment (icecream78) [reply api/mail.go:39]: @awalias you are absolutely right. This code have to update db field confirmation_sent_at, not confirmed_at. Thanks for pointing me at this fail =) I'll send fixed code

**Comment by icecream78** [comment]
@awalias is there any other questions about that changes? somewhy we stopped disscussion

**Pull request merged**